### PR TITLE
Add example for PyGTK with Matplotlib and Numpy

### PR DESCRIPTION
**Pull request by tobias47n9e**
I would like to add this example to the repository, that shows how to package pygtk, matplotlib and numpy. It is based on the pygtk-example and an example from the matplotlib-documentation. The whole example seems to only work with 32-bit and Python 2.7.x (Although a more experienced programmer might be able to debug this).

Compared to the other examples this one requires 7zip. I don't know how to do a requirements-check for this in the shell script, so it probably fails if `sh grab_files.sh` is executed on a system without 7zip.

pynsist_example_pygtk_matplotlib_numpy

### CODE REVIEW

File: examples/pygtk_mpl_numpy/README.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+used for extraction, but will leave the downloaded archives intact in the
+example directory.
+
+Matplotlib
+----------
+
+The example downloads the 32-bit Python 2.7 bindings of Matplotlib 1.4.3
+(matplotlib-1.4.3.win32-py2.7.exe).
+
+In the :code:`installer.cfg` Matplotlib additionally requires the sig, dateutil
```
Comment (takluyver) [inline examples/pygtk_mpl_numpy/README.rst]: s/sig/six/

File: examples/pygtk_mpl_numpy/README.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+Installer.cfg
+-------------
+
+The example is customized for 32-bit and Python 2.7. This is expressed in the
+:code:`installer.cfg`-file like this:
+
+::
+
+    version=2.7.7
```
Comment (takluyver) [inline examples/pygtk_mpl_numpy/README.rst]: Latest is 2.7.9 - is there a reason to use 2.7.7?

File: examples/pygtk_mpl_numpy/grab_files.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Download the necessary files
+wget -O gtkbundle.zip http://ftp.gnome.org/pub/gnome/binaries/win32/gtk+/2.24/gtk+-bundle_2.24.10-20120208_win32.zip
+wget -O pygobject.exe http://ftp.gnome.org/pub/GNOME/binaries/win32/pygobject/2.28/pygobject-2.28.3.win32-py2.7.exe
+wget -O pycairo.exe http://ftp.gnome.org/pub/GNOME/binaries/win32/pycairo/1.8/pycairo-1.8.10.win32-py2.7.exe
+wget -O pygtk.exe http://ftp.gnome.org/pub/GNOME/binaries/win32/pygtk/2.24/pygtk-2.24.0.win32-py2.7.exe
+#Numpy and Matplotlib are both the bindings for Python 2.7 and 32-bit, but will work on a 64-bit Windows
```
Comment (takluyver) [inline examples/pygtk_mpl_numpy/grab_files.sh]: FYI, this is true of all Python extension modules - they need to match the bit-ness of the Python interpreter they're loaded by, but not necessarily the underlying OS. (though you can't run 64-bit Python on a 32-bit system)

**Comment by takluyver** [comment]
Thanks! I don't want to grow examples for every combination of packages, but it's useful to have at least one showing how to integrate matplotlib and numpy. I've made a couple of minor comments inline, but this basically looks fine.

I'm starting to think about having some kind of collection of either:
- Recipes for getting and unpacking these Windows installers automatically, or
- The files from these installers repacked in a standard format that can be automatically downloaded

Though if more projects take to shipping pre-compiled wheels, the need for this might reduce. I don't see the major GUI frameworks picking up wheels any time soon, though. Any thoughts are welcome.

BTW, is it [innsbruck-stereographic](https://github.com/tobias47n9e/innsbruck-stereographic) that you're working towards packaging? I don't pretend to understand it, but I'm interested in what people are trying to use Pynsist with.

File: examples/pygtk_mpl_numpy/grab_files.sh (hunk 3 above)
Comment (tobias47n9e) [inline examples/pygtk_mpl_numpy/grab_files.sh]: That sounds reasonable. I will remove that comment.

**Comment by tobias47n9e** [comment]
A more general guide to packaging would probably be a good idea. There are some other projects working on that, but I always find them difficult, because they never cover GUIs and they are usually programs with PyPI-only dependencies.

And yes, I would really like to package InnStereo (or Innsbruck Stereographic) for Windows. But I think it is a lot more complicated because it has more dependencies and it is Python 3.4, and as you know, depends on PyGObject.

But I am also thinking of adding pynsist in the packaging guide of my scientific-application cookbook (once the PyGObject stuff is sorted out): https://github.com/tobias47n9e/GTK3-Matplotlib-Cookbook

**Comment by takluyver** [comment]
Yes, as I see it, most of the tools and information out there are about packaging Python libraries, not applications. This is my attempt at a sane approach for Windows (I'm working on something for Linux), but this aspect - dealing with all the compiled packages - is still pretty rough at present.

I'll try to find time soon to fiddle with PyGObject - I'm sure it's possible to make it work. But it won't be this week - we're having the twice-yearly meeting to decide what to do next with IPython, which will take up all my time.

For cross platform GUIs, I would definitely recommend Qt over GTK - I definitely get the feeling that GTK (or at least the Python interfaces) is mainly made for Linux and Windows/Mac support is something of an extra. The different platforms feel much more equal in Qt.

**Comment by tobias47n9e** [comment]
Thank you again for taking the time to look at PyGObject. I have read before that Qt is a little better with cross-platform support, but I really like the Glade/PyGObject workflow (and the Gnome ecosystem). I also tend to think that the scientific community will converge towards Linux, so that the other platforms will not be as important in the future.

Good luck for the IPython meeting!

**Comment by takluyver** [comment]
> I also tend to think that the scientific community will converge towards Linux,

I wish I shared your optimism, but from everything I see, we seem to be converging on Macs for the time being. Among non-programmer-y scientists, I think there's still more people using Windows than Linux. I guess you're taking the long term view ;-)

Thanks!

**Pull request merged**